docs/source/newsfragments/4341.change.rst

```diff
@@ -0,0 +1 @@
+:func:`~cocotb.triggers.Join` and :meth:`.Task.join` now return the Task to be joined instead of a Trigger. This preserve backwards compatibility for ``await join`` or using the result in :class:`~cocotb.triggers.First` or :class:`~cocotb.triggers.Combine`.
```

Comment:
```suggestion
:func:`~cocotb.triggers.Join` and :meth:`.Task.join` now return the Task to be joined instead of a Trigger. This preserves backwards compatibility for ``await join`` or using the result in :class:`~cocotb.triggers.First` or :class:`~cocotb.triggers.Combine`.
```